lib/Sema/TypeCheckType.cpp

```diff
@@ -6097,13 +6128,32 @@ class ExistentialTypeSyntaxChecker : public ASTWalker {
       return;
     }
 
+    // Backtrack the stack, looking just through parentheses and metatypes. If
+    // we find an inverse (which always requires `any`), diagnose it specially.
```

Comment:
You might be able to just pass down the outermost location instead of "backtracking" here, but it's fine the way it is

Reply:
Thanks for sharing your thoughts on this, Slava. 

> You might be able to just pass down the outermost location

Unless you mean something

Reply:
Thanks for sharing your thoughts on this, Slava. 

> You might be able to just pass down the outermost location

Is this roughly what you meant (pass `tildeLoc` down such that if valid here, then the `~` necessarily pertains to the given identifier type)?

```c++
// walkToTypeReprPre

if (auto *inverse = isa<InverseTypeRepr>(T)) {
  this->tildeLoc = inverse->getTildeLoc();
} else if (this->tildeLoc.isValid()) {
  if (!T->isParenType() &&
      !isa<MetatypeTypeRepr>(T) &&
      !isa<DeclRefTypeRepr>(T)) {
    this->tildeLoc = SourceLoc();
  }
}

// ...
// checkDeclRefTypeRepr(declRefTR);

this->tildeLoc = SourceLoc();
```

Reply:
I’m going to try to get this into the release as is before it’s too late, but I’m still very much willing to discuss this alternative.